### ISSUE TITLE
Starting sequence block / page indices in decode from 0 causes -inf logits in IREE Toy Llama

### ISSUE THREAD

**Issue by sogartar** [nod-ai/shark-ai#2355]
I just noticed this when I decided to change [here](https://github.com/nod-ai/shark-ai/blob/bdbf19143901d2008a5e2c57a0589a9eaaf292f7/sharktank/sharktank/utils/llm_utils.py#L290)
```
pages = numpy.arange(start=1, stop=bs * count + 1, dtype=numpy.int64)
```
to
```
pages = numpy.arange(start=0, stop=bs * count, dtype=numpy.int64)
```

The test [tests/models/llama/toy_llama_test.py::TestToyLlamaIree::testDecodePerplexity](https://github.com/nod-ai/shark-ai/blob/bdbf19143901d2008a5e2c57a0589a9eaaf292f7/sharktank/tests/models/llama/toy_llama_test.py#L137)
fails with
```
FAILED sharktank/tests/models/llama/toy_llama_test.py::TestToyLlamaIree::testDecodePerplexity[False] - assert False
 +  where False = LlmPerplexityEval.Result(valid=False, score=0.8056640625).valid
```

In this scenario decode returns a bunch of `inf` logits.
I don't understand why 0 is special. It should not be. In eager all is fine.

**Comment by sogartar** [comment]
cc @rsuderman
